test/lib/list.test.js

```diff
@@ -3,6 +3,28 @@
 var expect = require('chai').expect;
 var list   = require('../../lib/list');
 
+function testListProgress (file, done) {
+  var error;
+  list('test/zip.7z')
+  .progress(function (entries) {
+    try {
+      expect(entries.length).to.be.at.least(1);
+      entries.forEach(function (entry) {
+        expect(entry.date).to.be.an.instanceof(Date);
+        expect(entry.attr.length).to.eql(5);
+        expect(entry.name).to.be.a('string');
+        expect(entry.name).to.not.contain('\\');
+      });
+    } catch(e) {
+      error = e;
+    }
+  })
+  .then(function () {
+    done(error)
+  })
+  .done(next);
```

Comment:
The reason I made this little try-catch mess is that the errors from the `expect` statements weren't being shown correctly in the console due to them being called asynchronically in `progress`